Ajout d'un systeme modulaire

jbeaudoin11 opened the pull request:
J'ai créer un systeme qui permet de loader des parties spécifique pour
chaque page. Donc il serait plus facile dans l'avenir de modifier les
autres pages. je n'ai pas fini donc n'accept pas la pull request. C'est seulement pour indication

Commit:
Ajout d'un systeme modulaire

J'ai créer un systeme qui permet de loader des parties spécifique pour
chaque page. Donc il serait plus facile dans l'avenir de modifier les
autres pages

jbeaudoin11 commented:
J'ai aussi fais de la traduction... Désoler, mais le code en français ça n'existe pas!
Il en manque un peu..

antoineMoPa commented:
Nice le système modulaire!

Pour le français, j'ai rushé a accepter parce que c'est seulement pour l'UdeS. Mais je me suis dit que si l'extension fini par améliorer moodle (et autres), des parties du code pourraient être utilisées dans d'autres universitées. (Le code aurait donc intérêt à être en anglais au complet)

Je vais faire un pull (en local, pas sur github) en fin de semaine voir ce que ça donne et probablement merger le tout!

Commit:
Changement dans le calculs des notes

J'ai changer le code dans le calculs des notes pour rendre ça plus
simple a comprendre et j'ai ajouté la colonne TÉ

Commit:
Petit changement dans le makeFile

J'ai enlever la copie dans le dossier chrome vu que pour l'instant je
crois qu'on se concentre sur Firefox.

Commit:
Merge remote-tracking branch 'refs/remotes/antoineMoPa/master'

Commit:
Petit ajout

quelques commentaires

Commit:
Amélioration de la logique derrière la calcul des notes

J'ai fais une fonction qui créer un objet avec les data importante de
toutes les notes. Plus facile a manipuler !

Commit:
Rajout du header dans l'objet data

jbeaudoin11 commented:
J'ai changer pas mal de chose.

J'ai travailler sur l'abstraction des données. Donc j'ai un objet data qui contient le header (layout) avec les 3 collones de prefix ainsi que les groups GENXXX et leur taille respective. Par la suite j'ai aussi les rows qui contient toutes les paramètres.

bref c'est beaucoup plus simple maintenant qu'on a toutes les data du tableau sous la forme d'un objet on peu changer l'interface.

Demain/tanto je vais probablement travailler sur ça justement, l'interface. Je pensais utiliser bootstrap comme css, ça te convient ? Le seule problème c'est que le tableau va etre beau et le reste laid en tbk xD

antoineMoPa commented:
Ça marche! Bootstrap aussi.

Faut juste qu'on fasse marcher ce que tu fais pour firefox dans chrome. Alors essaye de garder ce qui est juste pour firefox (require et les trucs du SDK) dans des fichiers séparés.

antoineMoPa commented:
J'ai mergé dans mon repo vu que ça commençait à être vraiment différent de l'original.

Par contre, va falloir qu'on s'organise pour faire marcher les choses dans chrome. Faudrait pas pelleter trop ça par en avant sinon ça va être dur de faire une prochaine release.

antoineMoPa commented:
Pour les données, c'est vraiment une bonne affaire ce que tu as fait, c'était assez mal organisé dans les données brutes.
